### ISSUE TITLE
use subcommand instead of so many commands

### ISSUE THREAD

**Issue by jianshu93** [lskatz/fasten#36]
Hi fasten team,

In some cases where I have limited disk space for default rust  binary install path, I have to compile it and put binary into a user path, I find it to be not so convenient for so many commands. Is it possible to use subcommand for each commands, just like what seqkit did, I feel this way it is more natural. Also clap 4.5 can be use for such a case. 

Thanks,

Jianshu

**Comment by lskatz** [comment]
Hi, this seems like a good idea but it would be a huge overhaul. Do you feel like the executables take too much space? It seems to be at about 35M for me which is not very large considering the size of fastq.gz files that Fasten would be analyzing.

**Comment by tseemann** [comment]
I agree with @jianshu93 - best practice for command line tools in our field are to use sub commands. A single binary Swiss Army Knife. Unclear why it would a _huge_ overhaul. Maybe use Co-pilot to do it for you :-P
